starboard/android/shared/media_decoder.cc

```diff
@@ -638,12 +683,138 @@ void MediaDecoder::OnMediaCodecOutputFormatChanged() {
   ScopedLock scoped_lock(mutex_);
   dequeue_output_results_.push_back(dequeue_output_result);
   condition_variable_.Signal();
+
+  // TODO(borongchen): remove below, only for debugging
+  /*SB_LOG(ERROR) << "Brown FormatChanged"
+                << ", dequeue_output_results_: " <<
+     dequeue_output_results_.size()
+                << ", pending_flush_count_: "
+                << pending_flush_count_.load()
+                << ", pending_tasks_: " << pending_tasks_.size();*/
 }
 
 void MediaDecoder::OnMediaCodecFrameRendered(int64_t frame_timestamp) {
   frame_rendered_cb_(frame_timestamp);
 }
 
+bool MediaDecoder::Reset() {
+  // If it is end of the stream, return |false| to ecreate the codec completely.
+  if (stream_ended_.load()) {
+    return false;
+  }
+
+  // If it is not end of the stream, then try to flush if we can,
+  // otherwise return |false| to recreate the codec completely.
+  if (!force_flush_during_seek_ || !is_valid() || !TryFlush()) {
```

Comment:
Removed `is_valid()` as we already have this check in `TryFlush()`. I didn't add `is_valid()` in `TryFlush()` at the beginning, but since we have this right now, removed `is_valid()` here.